### ISSUE TITLE
bad-whitespace incorrectly reported for type hinted kwarg

### ISSUE THREAD

**Issue by alexhayes** [PyCQA/pylint#1507]
Firstly I'd preface this with I'm not 100% sure of the correct behaviour. It would seem from from [PEP-484](https://www.python.org/dev/peps/pep-0484/#instantiating-generic-classes-and-type-erasure) that perhaps PyLint has got this correct but I thought I'd raise it to be sure.

### Steps to reproduce
1. Define a valid Python 3.5+ function like so;
    ```python
    def check(first: int=1):
         pass
    ```
2. Run pylint (ensuring that `bad-whitespace` / `C0326` is not disabled.

### Current behavior

The following is incorrectly reported;

```
[bad-whitespace (C0326)] Exactly one space required around keyword argument assignment
def check(first: int=1):
                    ^
````

### Expected behavior

No errors reported from PyLint.

### pylint --version output

Latest from master for both pylint and astroid as of 2017-05-28.

```
pylint 1.8.0,
astroid 1.5.2
Python 3.6.0 (default, Jan  6 2017, 15:54:20)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

**Comment by PCManticore** [comment]
I think pylint is incorrect, since PEP 8 states that ``Don't use spaces around the = sign when used to indicate a keyword argument or a default parameter value.``

**Comment by hikoz** [comment]
According to #1421, this issue is incorrect.

**Comment by alexhayes** [comment]
**TL/DR**

PEP-8 actually does specifically mention the use of spaces around default parameters.

> When combining an argument annotation with a default value, use spaces around the = sign (but only for those arguments that have both an annotation and a default).
> 
> Yes:
> ```python
> def munge(sep: AnyStr = None): ...
> def munge(input: AnyStr, sep: AnyStr = None, limit=1000): ...
>
> No:
> ```python
> def munge(input: AnyStr=None): ...
> def munge(input: AnyStr, limit = 1000): ...
> ```

**More Info**

Issue #1421 points to [PEP-3107](https://www.python.org/dev/peps/pep-3107/#syntax) which has two different uses, one with spaces and one without. Given, the usage without spaces uses parentheses and thus complicates the matter but point being there is no mention of a space being required in PEP-3107.

PEP-484 also makes no mention of spaces around default parameters, however it does have examples of doing exactly that.
